portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,32 +29,39 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
     public ArrayList<String> list = new ArrayList<String>();
+    public String lang;
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
     //Query
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query = new Query("Comment");
 
     PreparedQuery results = datastore.prepare(query);
-
-    List<Comment> comments = new ArrayList<>();
+    ArrayList<Comment> comments = new ArrayList<Comment>();
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      String comment = (String) entity.getProperty("text");
-
-      Comment msg = new Comment(id, comment);
-      comments.add(msg);
+        long id = entity.getKey().getId();
+        String comment = (String) entity.getProperty("text");
+        if(lang != null && comment != null){
+            Translate translate = TranslateOptions.getDefaultInstance().getService();
+            Translation translation = translate.translate(comment, Translate.TranslateOption.targetLanguage(lang));
+            String translatedText = translation.getTranslatedText();
+            comment = translatedText;
+        }
+        Comment msg = new Comment(id, comment);
+        comments.add(msg);
     }
-
-    Gson gson = new Gson();
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(comments));
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
+    //Get translate language
+    lang = request.getParameter("languages");
```

Comment:
I understand this is here because the form is written using a POST request. I think a cleaner way to do this (doesn't have to be for this PR, but wanted to point this out) is to have this in GET because the language param has more to do with what language you want to _get_ the comments in. A potential solution is to create separate form for just the languages option with method = GET instead of POST (though see https://www.w3schools.com/tags/att_form_method.asp) in order to send the language param and have it be received by doGet().

Reply:
Yeah I think I am still getting a grasp on the concept of POST and GET, I will try to implement this during my spring break when I am all caught up. Thank you for clarifying!